packages/core/framework/src/http/utils/refetch-entities.ts

```diff
@@ -1,49 +1,70 @@
-import { MedusaContainer } from "@medusajs/types"
-import {
-  ContainerRegistrationKeys,
-  isString,
-  remoteQueryObjectFromString,
-} from "@medusajs/utils"
-import { MedusaRequest } from "../types"
+import type {
+  GraphResultSet,
+  MedusaContainer,
+  RemoteJoinerOptions,
+  RemoteQueryFunctionReturnPagination,
+} from "../../types"
+import { ContainerRegistrationKeys, isString } from "../../utils"
+import type { MedusaRequest } from "../types"
 
-export const refetchEntities = async (
-  entryPoint: string,
+export const refetchEntities = async <TEntry extends string>(
+  entryPoint: TEntry,
   idOrFilter: string | object,
   scope: MedusaContainer,
   fields: string[],
   pagination?: MedusaRequest["queryConfig"]["pagination"],
-  withDeleted?: boolean
-) => {
-  const remoteQuery = scope.resolve(ContainerRegistrationKeys.REMOTE_QUERY)
-  const filters = isString(idOrFilter) ? { id: idOrFilter } : idOrFilter
-  let context: object = {}
+  withDeleted?: boolean,
+  options?: RemoteJoinerOptions
+): Promise<
+  Omit<GraphResultSet<TEntry>, "metadata"> & {
+    metadata: RemoteQueryFunctionReturnPagination
+  }
+> => {
+  const query = scope.resolve(ContainerRegistrationKeys.QUERY)
+  let filters = isString(idOrFilter) ? { id: idOrFilter } : idOrFilter
+  let context!: Record<string, unknown>
 
   if ("context" in filters) {
-    if (filters.context) {
-      context = filters.context!
+    const { context: context_, ...rest } = filters
+    if (context_) {
+      context = context_! as Record<string, unknown>
     }
-
-    delete filters.context
+    filters = rest
   }
 
-  const variables = { filters, ...context, ...pagination, withDeleted }
-
-  const queryObject = remoteQueryObjectFromString({
-    entryPoint,
-    variables,
+  const graphOptions: Parameters<typeof query.graph>[0] = {
+    entity: entryPoint as string,
     fields,
-  })
+    filters,
+    pagination,
+    withDeleted,
+    context: context,
+  }
 
-  return await remoteQuery(queryObject)
+  return (await query.graph(graphOptions, options)) as Omit<
```

Comment:
Self-review: set metadata to empty object to prevent undefined to pass through and maintain some level of compatibility